### DISCUSSION TITLE
Any reason why Expressions and If Statements are not enabled by default?

### DISCUSSION THREAD

**Discussion by primaltechguides** [arendst/Tasmota#16683] [Q&A]
Just wondering what the reason is for not having the below features enabled as default? Do they use so much extra space?

#define USE_EXPRESSION
#define SUPPORT_IF_STATEMENT 

I've recently used expressions and if statements to control a 3 Gang switch. I had to go through compiling in order to achieve this.

**Comment by sfromis** [comment]
The general structure is that Tasmota has so many features that ESP8266 just does not have room for enabling "everything". And expressions/if is not needed for the large majority devices. Many of the potential use cases can be done without, with a bit of extra tricks.
